web/pgadmin/static/js/AppMenuBar.jsx

```diff
@@ -73,6 +75,12 @@ export default function AppMenuBar() {
   }, []);
 
   const getPgMenuItem = (menuItem, i)=>{
+    let shortcut;
+    // Fetch shortcut from preferences if defined in menuItem.
+    if(menuItem.shortcut_preference) {
+      const [module, key] = menuItem.shortcut_preference;
+      shortcut = prefStore.getPreferences(module, key)?.value;
```

Comment:
Lets not fetch shortcut on every render. It won't change very frequently. Keep it in all_menus_cache and update on all_menus_cache on preference change.